libraries/CMakeLists.txt

```diff
@@ -1,16 +1,42 @@
-# Add build configuration for all 3rd party modules.
-add_subdirectory(${3RDPARTY_DIR})
-
-# Macro utility to clone the coreMQTT submodule.
-macro( clone_mqtt )
-        find_package( Git REQUIRED )
-        message( "Cloning submodule coreMQTT." )
-        execute_process( COMMAND rm -rf ${MODULES_DIR}/standard/coreMQTT
-                         COMMAND ${GIT_EXECUTABLE} submodule update --init --recursive libraries/standard/coreMQTT
-                         WORKING_DIRECTORY ${ROOT_DIR}
-                         RESULT_VARIABLE MQTT_CLONE_RESULT )
-
-        if( NOT ${MQTT_CLONE_RESULT} STREQUAL "0" )
-                message( FATAL_ERROR "Failed to clone coreMQTT submodule." )
+# Macro utility to clone a library and source-related submodules.
+macro( clone_library submodule_dir clone_source )
+    find_package( Git REQUIRED )
+    if( NOT EXISTS ${submodule_dir}/.git )
+        # Attempt to clone submodule.
+        if( ${BUILD_CLONE_SUBMODULES} )
+            execute_process( COMMAND rm -rf ${submodule_dir}
+                        COMMAND ${GIT_EXECUTABLE} submodule update --init ${submodule_dir}
+                        WORKING_DIRECTORY ${ROOT_DIR}
+                        RESULT_VARIABLE LIB_CLONE_RESULT )
+            if( NOT ${LIB_CLONE_RESULT} STREQUAL "0" )
+                message( FATAL_ERROR "Failed to clone submodule ${submodule_dir}." )
+            endif()
+            if(${clone_source})
+                # Sources may have submodules like from the depedency directory
+                execute_process( COMMAND ${GIT_EXECUTABLE} submodule update --init -- source
```

Comment:
This carries the knowledge about `source` directory of the individual repositories owned by us, and somewhat mixes it with the third-party repositories for which `source` is not relevant (thereby, requiring the need of providing the `0` parameter). I do not this that this commonization is clean.
Instead, it would be cleaner to just have this file update all submodules (owned by us) at once with the `--init --recursive` flag. As our individual submodules have been updated to disable cloning of the CBMC and Unity/Cmock leaf submodules, using the `--recursive` flag will be clean here.